File: 20-ValidParentheses/20-ValidParentheses.java
// Last updated: 7/14/2026, 2:06:07 PM
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st=new Stack<>();
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='{'){
                st.push('}');
            }else if(s.charAt(i)=='('){
                st.push(')');
            }else if(s.charAt(i)=='['){
                st.push(']');
            }else if(st.isEmpty()||s.charAt(i)!=st.pop()){
                return false;
            }
        }return st.isEmpty();
    }
}

File: 20-ValidParentheses/20-ValidParentheses.py
# Last updated: 7/14/2026, 2:06:02 PM
class Solution:
    def isValid(self, s: str) -> bool:
        l=[]
        m={'(':')','[':']','{':'}'}
        for i in s:
            if i in m.keys():
                l.append(i)
            elif i in m.values():
                if l == [] or m[l.pop()] != i:
                    return False
            else:
                return False
        return len(l)==0

        